### PR TITLE
Added rules to remove unsupported compiler flags

### PR DESCRIPTION
**Pull request by TejaX-Alaghari**
Added CMake rules to remove unsupported compiler flags
- prec_div
- parallel
- wd[warnings]

### CODE REVIEW

File: clang/test/dpct/cmake_migration/case_021/input.cmake [hunk 1]
```diff
@@ -11,3 +11,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
 add_link_options("--add-arch=sm_70")
 list(APPEND PMEMD_NVCC_FLAGS --std c++11)
+
+string(APPEND CMAKE_C_FLAGS " -prec_div")
+string(APPEND CMAKE_CXX_FLAGS " -parallel")
+string(APPEND CMAKE_CXX_FLAGS " -wd188,186,144,913,556,858,597,177,1292,167,279,592,94,2722,3199")
```
Comment (tomflinda) [inline clang/test/dpct/cmake_migration/case_021/input.cmake]: 1. These three options are not NVCC-specific options,  which compiler uses those options?
2. In my opinion, if it is only used in a specific application for a specific compiler, we can implement those migration rules as application-specific rules?

Comment (TejaX-Alaghari) [reply clang/test/dpct/cmake_migration/case_021/input.cmake]: 1. These options are used by Intel Classical C Compiler (ICC) - [ref](https://www.intel.com/content/www/us/en/docs/cpp-compiler/developer-guide-reference/2021-8/alphabetical-option-list.html)
2. I haven't found any other examples of apps using these flags yet

Comment (tomflinda) [reply clang/test/dpct/cmake_migration/case_021/input.cmake]: Then,  I suggest that those migration rules should be implemented as application-specific rules.

Comment (zhimingwang36) [reply clang/test/dpct/cmake_migration/case_021/input.cmake]: It's kind of migration knowledge, how about put it into a separate cmake_script_migration_rule_optional.yaml?

File: clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml [hunk 2]
```diff
@@ -3175,3 +3175,42 @@
       MatchMode: Full
       In: \${CMAKE_CUDA_RUNTIME_LIBRARY_LINK_OPTIONS_SHARED}
       Out: ""
+
+- Rule: rule_unsupported_flags_prec_div
```
Comment (zhimingwang36) [inline clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml]: Please remove this,  as  it is already in [make_script_migration_rule_optional.yaml].

Comment (TejaX-Alaghari) [reply clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml]: I removed these rules but they appeared again while rebasing.
Removed now.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

File: clang/tools/dpct/DpctOptRules/cmake_script_migration_rule_optional.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Copyright (C) Intel Corporation
```
Comment (tomflinda) [inline clang/tools/dpct/DpctOptRules/cmake_script_migration_rule_optional.yaml:1]: Why add a new file "clang/tools/dpct/DpctOptRules/cmake_script_migration_rule_optional.yaml"?  It is NOT auto-loaded by SYCLomatic during migrating CMake scripts.

File: clang/test/dpct/cmake_migration/case_066/input.cmake [hunk 4]
```diff
@@ -0,0 +1,3 @@
+string(APPEND CMAKE_C_FLAGS " -prec_div")
+string(APPEND CMAKE_CXX_FLAGS " -parallel")
+string(APPEND CMAKE_CXX_FLAGS " -wd188,186,144,913,556,858,597,177,1292,167,279,592,94,2722,3199")
```
Comment (tomflinda) [inline clang/test/dpct/cmake_migration/case_066/input.cmake:3]: These three options are not nvcc-specific options,  which compiler can support these three options?

File: clang/tools/dpct/DpctOptRules/cmake_script_migration_rule_optional.yaml (hunk 3 above)
Comment (zhimingwang36) [reply clang/tools/dpct/DpctOptRules/cmake_script_migration_rule_optional.yaml:1]: Yes, this the reason why the PR add the rule into a sperate file.

Comment (tomflinda) [reply clang/tools/dpct/DpctOptRules/cmake_script_migration_rule_optional.yaml:1]: Okay

File: clang/test/dpct/cmake_migration/case_066/input.cmake (hunk 4 above)
Comment (TejaX-Alaghari) [reply clang/test/dpct/cmake_migration/case_066/input.cmake:3]: Its the old ICC compiler as mentioned in [this](https://github.com/oneapi-src/SYCLomatic/pull/2369#discussion_r1768408482) comment

**Review by tomflinda — APPROVED** [review]
LGTM

**Pull request merged**